include/zephyr/acpi/acpi.h

```diff
@@ -15,6 +15,27 @@
 #define ACPI_DMAR_FLAG_X2APIC_OPT_OUT			BIT(1)
 #define ACPI_DMAR_FLAG_DMA_CTRL_PLATFORM_OPT_IN	BIT(2)
 
+#define ACPI_MMIO_GET(res) (res)->reg_base[0].mmio
+#define ACPI_IO_GET(res) (res)->reg_base[0].port
+#define ACPI_RESOURCE_SIZE_GET(res) (res)->reg_base[0].length
+#define ACPI_RESOURCE_TYPE_GET(res) (res)->reg_base[0].type
+
+#define ACPI_MULTI_MMIO_GET(res, idx) (res)->reg_base[idx].mmio
+#define ACPI_MULTI_IO_GET(res, idx) (res)->reg_base[idx].port
+#define ACPI_MULTI_RESOURCE_SIZE_GET(res, idx) (res)->reg_base[idx].length
+#define ACPI_MULTI_RESOURCE_TYPE_GET(res, idx) (res)->reg_base[idx].type
+
+#define ACPI_RESOURCE_COUNT_GET(res) (res)->mmio_max
+
+enum acpi_res_type {
+	/** IO mapped Resource type */
+	ACPI_RES_TYPE_IO = 1,
```

Comment:
Ok, I took a look at how you're using these, and it seems the numeric values don't matter. Because of that the `= 1` is misleading, since it gives the impression that it's important that `TYPE_IO` has exactly the value `1`. I'd propose to just remove the `= 1` and let the values be auto-assigned (i.e. starting from `0`).